DMCompiler/DM/Visitors/DMProcBuilder.cs

```diff
@@ -414,5 +415,49 @@ class DMProcBuilder {
             DMExpression.Emit(_dmObject, _proc, statementOutputControl.Control);
             _proc.OutputControl();
         }
+
+        public void ProcessStatementTryCatch(DMASTProcStatementTryCatch tryCatch) {
+            string catchLabel = _proc.NewLabelName();
+            string endLabel = _proc.NewLabelName();
+
+            _proc.StartScope();
+            ProcessBlockInner(tryCatch.TryBody);
+            _proc.EndScope();
+            _proc.Jump(endLabel);
+
+            //TODO handle these properly instead of being copypasted from DMASTProcDefinition
+            if (tryCatch.CatchParameters != null)
+            {
+                foreach (DMASTDefinitionParameter parameter in tryCatch.CatchParameters) {
+                    string parameterName = parameter.Name;
+
+                    _proc.AddLocalVariable(parameterName, parameter.ObjectType);
+                    if (parameter.Value != null) {
+                        string afterDefaultValueCheck = _proc.NewLabelName();
+
+                        _proc.PushLocalVariable(parameterName);
+                        _proc.IsNull();
+                        _proc.JumpIfFalse(afterDefaultValueCheck);
+
+                        _proc.PushLocalVariable(parameterName);
+                        try {
+                            DMExpression.Emit(_dmObject, _proc, parameter.Value, parameter.ObjectType);
+                        } catch (CompileErrorException e) {
+                            Program.Error(e.Error);
+                        }
+                        _proc.Assign();
+
+                        _proc.AddLabel(afterDefaultValueCheck);
+                    }
+                }
+            }
```

Comment:
I tested this code in byond:
```
try
	throw null
catch(var/exception/E = 3)
	usr << isnull(E)
	usr << E
```
This compiled fine, and output:
```
1

```
It appears that default values don't do anything